### PR TITLE
Home screen + Upfront Habit Event Addition

### LINKED ISSUE #10: 01.07.01 Add a page to view daily habits
**User statement**
As a doer, I want to see what habits I need to do today.

**Describe the solution you'd like**
Add activity/view that shows the habits to complete today.

**Development notes**

### PR DESCRIPTION
**Pull request by Nova-Drive**
This PR adds a home screen of the habits to do today, as well as a means to quickly add a new habit event for each habit.

This PR closes #10 and #88.

### CODE REVIEW

File: app/src/androidTest/java/com/example/habitsmasher/MainActivityTest.java
Comment (Nova-Drive) [inline app/src/androidTest/java/com/example/habitsmasher/MainActivityTest.java:338]: This test ensures that when a habit that is supposed to take place on the current day is added, it shows up on the home screen.

File: app/src/androidTest/java/com/example/habitsmasher/MainActivityTest.java
Comment (Nova-Drive) [inline app/src/androidTest/java/com/example/habitsmasher/MainActivityTest.java:397]: This test ensures that when a habit that is **not** supposed to take place on the current day is added, it does **not** show up on the home screen.

File: app/src/androidTest/java/com/example/habitsmasher/MainActivityTest.java
Comment (Nova-Drive) [inline app/src/androidTest/java/com/example/habitsmasher/MainActivityTest.java:1053]: This test ensures that the core functionality of adding a habit event from the home screen works. In theory it should be unaffected, since we navigate to the habit event list fragment and all the work is done through there.

File: app/src/androidTest/java/com/example/habitsmasher/MainActivityTest.java
Comment (Nova-Drive) [inline app/src/androidTest/java/com/example/habitsmasher/MainActivityTest.java:1640]: Helpers for setting the current day. Important since main implementation checks this.

File: app/src/androidTest/java/com/example/habitsmasher/MainActivityTest.java
Comment (Nova-Drive) [inline app/src/androidTest/java/com/example/habitsmasher/MainActivityTest.java:1652]: gets the current day. Also appears in the home screen fragment.

File: app/src/main/java/com/example/habitsmasher/DaysTracker.java
Comment (Nova-Drive) [inline app/src/main/java/com/example/habitsmasher/DaysTracker.java:81]: There's a new way of setting the days of the week: ArrayList of strings. This is needed to conform to the way we populate the lists of habits (Through the queries to the DB). Now we just query based on whether the current day shows up in the array of days in the DB. I'm keeping the string based constructors and the like, however.

File: app/src/main/java/com/example/habitsmasher/DaysTracker.java
Comment (Nova-Drive) [inline app/src/main/java/com/example/habitsmasher/DaysTracker.java:137]: I split up the setDays method into two bits, to handle two different cases with one method. This makes it very convenient and scalable.

File: app/src/main/java/com/example/habitsmasher/Habit.java
Comment (Nova-Drive) [inline app/src/main/java/com/example/habitsmasher/Habit.java:44]: A lot of files are essentially just little changes like this to adapt to changes.

File: app/src/main/java/com/example/habitsmasher/Habit.java
Comment (Nova-Drive) [inline app/src/main/java/com/example/habitsmasher/Habit.java:18]: _days is now an arraylist of strings, doesn't change too much since you can just get the string of days by passing it through a DaysTracker object.

File: app/src/main/java/com/example/habitsmasher/HabitEventList.java
Comment (Nova-Drive) [inline app/src/main/java/com/example/habitsmasher/HabitEventList.java:56]: Found this while paroozing, changed since we don't use username anymore.

File: app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java
Comment (Nova-Drive) [inline app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java:169]: Habit still takes a string, it's converted in the habit constructor to a list.

File: app/src/main/java/com/example/habitsmasher/ui/dashboard/HomeHabitItemAdapter.java
Comment (Nova-Drive) [inline app/src/main/java/com/example/habitsmasher/ui/dashboard/HomeHabitItemAdapter.java:55]: This new adapter is needed to use the new home_habit_row I created. I found that you can't edit the row without making a new adapter, so that's what I did.

File: app/src/main/java/com/example/habitsmasher/ui/dashboard/HomeHabitItemAdapter.java
Comment (Nova-Drive) [inline app/src/main/java/com/example/habitsmasher/ui/dashboard/HomeHabitItemAdapter.java:63]: Pretty much the only difference between the old adaptor and this one.

File: app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java
Comment (Nova-Drive) [inline app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java:56]: This class is what runs the home fragment. It's also very similar to the list fragment, with a few changes to some logic and such.

File: app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java
Comment (Nova-Drive) [inline app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java:131]: Gets all the habits that take place on the current day. This is what all of the DaysTracker changes lead up to.

File: app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java
Comment (Nova-Drive) [inline app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java:197]: New swipe button is here, and I'll probably need to change this to fit in with the new listener abstractions and such.

File: app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java
Comment (Nova-Drive) [inline app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java:205]: Originally, I wanted to straight up open the dialog box and skip the list, but that was deemed impossible.

File: app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java
Comment (Nova-Drive) [inline app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java:241]: This gets the current days abbreviation, such as "WE".

File: app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java
Comment (Nova-Drive) [inline app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java:272]: Both of these are essentially just copied from other functions with the new action IDs.

File: app/src/main/res/values/strings.xml
Comment (Nova-Drive) [inline app/src/main/res/values/strings.xml:50]: set to "null" by default, edited in home fragment.

File: app/src/main/res/values/strings.xml
Comment (Nova-Drive) [inline app/src/main/res/values/strings.xml:51]: Keeping this here for future plan, will have rotating welcome message but for now this is unused.

**Comment by Nova-Drive** [comment]
Forgot to look at the unit tests, I'll get that sorted.

File: app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java
Comment (JuliePilz) [reply app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java:205]: I think you could probably remove these comments? If it's not being used you can also remove the function

**Review by JuliePilz — APPROVED** [review]
Nice job, this is great!!

**Review by rudy-patel — CHANGES_REQUESTED** [review]
Looks nice! You can take a look at some of the comments I've left. Any tasks (checkboxes) I've left are things I think should be completed **in this PR** before it's merged. I would also appreciate some screenshots to get a visual picture of what's happening. I also didn't see much documentation (function/class headers), so let's add that too!

- [ ] add missing javadoc documentation
- [ ] add screenshots

File: app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java
Comment (Nova-Drive) [reply app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java:205]: Tried getting rid of the function, I need to have it to satisfy the abstraction from ListFragment. I'll get rid of the comments though.

**Review by rudy-patel — APPROVED** [review]
Approved, the screenshots don't matter too much, just trusting you that everything looks good. I have spun up a task to add the missing `javadoc` comments in this PR () and assigned it to you so we don't forget about it + hold up this PR for the demo.

**Review by JacobNguyen71 — APPROVED** [review]

File: app/src/main/java/com/example/habitsmasher/ui/dashboard/HomeHabitItemAdapter.java
Comment (kaden-dreger) [reply app/src/main/java/com/example/habitsmasher/ui/dashboard/HomeHabitItemAdapter.java:79]: Agreed^

**Review by kaden-dreger — APPROVED** [review]
Looks good.

**Pull request merged**
